### PR TITLE
feat(ui): use msg_grid based implementation for cmdheight=0

### PR DESCRIPTION
**Pull request by bfredl**

### CODE REVIEW

**Comment by bfredl** [comment]
https://github.com/neovim/neovim/issues/20153 is not yet fixed

File: src/nvim/getchar.c [hunk 1]
```diff
@@ -1655,10 +1655,6 @@ static void getchar_common(typval_T *argvars, typval_T *rettv)
   no_mapping++;
   allow_keys++;
   for (;;) {
-    // Position the cursor.  Needed after a message that ends in a space,
-    // or if event processing caused a redraw.
-    ui_cursor_goto(msg_row, msg_col);
```
Comment (zeertzjq) [inline src/nvim/getchar.c:1660]: I think this breaks backward-compatibility.

Comment (bfredl) [reply src/nvim/getchar.c:1660]: fixed.

**Comment by bfredl** [comment]
> I just tried the latest commit today, and it caused nvim segfault immediately upon every start. I had to switch back to nightly.

It is a feature branch, it will merged to master once it is ready for nighty use.

Need a minimal repro for the segfault (and/or a traceback), otherwise it is not useful information.

**Pull request merged**